equinix/data_source_metal_device.go

```diff
@@ -232,26 +237,28 @@ func dataSourceMetalDeviceRead(d *schema.ResourceData, meta interface{}) error {
 	} else {
 		deviceId := deviceIdRaw.(string)
 		var err error
-		device, _, err = client.Devices.Get(deviceId, deviceReadOptions)
+		device, _, err = client.DevicesApi.FindDeviceById(nil, deviceId).Execute() // deviceReadOptions ?
```

Comment:
Includes have no impact here because the spec says that this comes back as an `Href`, so no matter what includes we send, `device.Project` always be an `Href`.  We cannot get this working nicely without improving the devices spec.  We've talked about adding `XOrHref` models across the board for this kind of issue, but IMO the experience of using those in code is not great; I have examples of that and of alternatively including `href` attributes directly in models, but haven't taken the time to open draft PRs in `metal-go` with those changes yet.  When they're up I will cross-link them with this PR.

Reply:
Draft PRs with the two modeling approaches mentioned above:

Reply:
The spec has been fixed and these properties are now accessible in `metal-go`.